### DISCUSSION TITLE
How to call a graphql endpoint using useFetch or useAsyncData

### DISCUSSION THREAD

**Discussion by SajidSedu** [nuxt/nuxt#24510] [Questions] [answered]
How do I  call a graphql endpoint using useFetch or useAsyncData, something like this:
```
const response = await axios.post(API_URL, {
        query: graphqlQuery,
        variables: { id: 'test_id' },
    });
```

**Comment by TheAlexLichter** [comment, marked as answer]
1. You can use a graphql client like https://github.com/Diizzayy/nuxt-graphql-client for that
2. You can do it raw via `useFetch(API_URL, { method: 'post', body: { query: graphqlQuery, vairables: { id: 'test_id' }}})`

**Reply by SajidSedu** [reply]
Thank you.
